src/commissaire/storage/client.py

```diff
@@ -0,0 +1,192 @@
+                '{}: Unable to save {} "{}": {}'.format(
+                    self.bus_mixin.__class__.__name__,
+                    model_instance.__class__.__name__,
+                    model_instance.primary_key, error))
+            raise error
+
+    def delete(self, model_instance):
+        """
+        Issues a "storage.delete" request over the bus using identifying
+        data from the model instance.
+
+        :param model_instance: Model instance with identifying data
+        :type model_instance: commissaire.models.Model
+        :raises: commissaire.bus.RemoteProcedureCallError
+        """
+        try:
+            params = {
+                'model_type_name': model_instance.__class__.__name__,
+                'model_json_data': model_instance.to_dict()
+            }
+            self.bus_mixin.request('storage.delete', params=params)
+        except RemoteProcedureCallError as error:
+            self.bus_mixin.logger.error(
+                '{}: Unable to delete {} "{}": {}'.format(
+                    self.bus_mixin.__class__.__name__,
+                    model_instance.__class__.__name__,
+                    model_instance.primary_key, error))
+            raise error
+
+    def list(self, model_class):
+        """
+        Issues a "storage.list" request over the bus for the given model
+        class, and returns a new model instance from the response data.
+
+        :param model_class: A concrete model class with a valid _list_class
+                            attribute
+        :type model_class: commissaire.models.Model
+        :returns: A new model instance
+        :rtype: model_class
+        :raises: commissaire.bus.RemoteProcedureCallError,
+                 commissaire.models.ValidationError
+        """
+        try:
+            assert model_class._list_attr is not None
+            assert model_class._list_class is not None
+            params = {
+                'model_type_name': model_class.__name__
+            }
+            response = self.bus_mixin.request('storage.list', params=params)
+            model_list = []
+            child_class = model_class._list_class
+            for child_dict in response['result']:
+                child_instance = child_class.new(**child_dict)
+                child_instance._validate()
+                model_list.append(child_instance)
+            model_instance = model_class.new()
+            setattr(model_instance, model_class._list_attr, model_list)
+            return model_instance
+        except (RemoteProcedureCallError, models.ValidationError) as error:
+            self.bus_mixin.logger.error(
+                '{}: Unable to list {}s: {}'.format(
+                    self.bus_mixin.__class__.__name__,
+                    model_class.__name__, error))
+            raise error
+
+    # -----------------------------------------------------------------------
+    # Model-specific methods (esp. listable types)
```

Comment:
Not needed, but I added them for convenience based on frequency of use.  There's a good many places were a host or cluster model is fetched with only an address or name string.